server/lib/ui-logger.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Wazuh app - Module for logging functions
+ * Copyright (C) 2015-2021 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+import winston from 'winston';
+import fs from 'fs';
+import { getConfiguration } from './get-configuration';
+import { WAZUH_DATA_LOGS_DIRECTORY_PATH, WAZUH_UI_LOGS_PLAIN_PATH ,WAZUH_UI_LOGS_RAW_PATH } from '../../common/constants';
+import { createDataDirectoryIfNotExists } from './filesystem';
+
+let allowed = false;
+let wazuhUiLogger : winston.Logger | undefined = undefined;
+let wazuhPlainUiLogger : winston.Logger | undefined = undefined;
+
+export interface IUIPlainLoggerSettings {
+    level: string,
+    message: string
+}
+
+export interface IUILoggerSettings extends IUIPlainLoggerSettings {
+    date: Date,
+    location: string
+}
+
+/**
+ * Here we create the loggers
+ */
+const initLogger = () => {
+  const configurationFile = getConfiguration();
+  const level =
+    typeof (configurationFile || {})['logs.level'] !== 'undefined' &&
+    ['info', 'debug'].includes(configurationFile['logs.level'])
+      ? configurationFile['logs.level']
+      : 'info';
+
+
+  wazuhUiLogger = winston.createLogger({
+    level,
+    format: winston.format.json(),
+    transports: [
+      new winston.transports.File({
+        filename: WAZUH_UI_LOGS_RAW_PATH
+      })
+    ]
+  });
+
+  // Prevents from exit on error related to the logger.
+  wazuhUiLogger.exitOnError = false;
+
+  wazuhPlainUiLogger = winston.createLogger({
+    level,
+    format: winston.format.simple(),
+    transports: [
+      new winston.transports.File({
+        filename: WAZUH_UI_LOGS_PLAIN_PATH
+      })
+    ]
+  });
+
+  // Prevents from exit on error related to the logger.
+  wazuhPlainUiLogger.exitOnError = false;
+};
+
+/**
+ * Checks if wazuh/logs exists. If it doesn't exist, it will be created.
+ */
+const initDirectory = async () => {
```

Comment:
move all common methods with `logger` to a shared support class to avoid code repetition